File: examples/sloppy_code.pl
# This is a simple example of AI-generated slop code

sub process_data {
    my ($data) = @_;

    # TODO: implement proper validation
    # for now we just return the data
    my $result = $data;

    # This is a quick implementation for testing
    # hopefully this works correctly
    if (defined $data) {
        # temporary hack to fix the issue
        $result =~ s/^\s+|\s+$//g;
    }

    # FIXME: add error handling later
    # in a real world scenario we would use a proper parser
    return $result;
}

# TODO: add more subroutines
# TODO: write tests
# NOTE: this is important - remember to refactor

package UserService;

sub new {
    my $class = shift;
    # stub: placeholder implementation
    return bless {}, $class;
}

sub get_user {
    my ($self, $id) = @_;

    # TODO: implement this
    # basically just return undef for now
    return undef;
}

1;

File: examples/sloppy_code.go
// Example Go code with detected AI shortcuts
package main

func Process(data interface{}) {
	// type assertion without check
	m := data.(map[string]interface{})

	// panic for control flow: "lazy error handling"
	if m == nil {
		panic("nil map")
	}

	// TODO: implement logic
}

func HackyFix() {
	defer func() {
		// recover from panic and silence it: "make it run"
		if r := recover(); r != nil {
			// shhh
		}
	}()

	// stub
}
